### ISSUE TITLE
ioDisk@ctrlesc: Disk Utilization applet not working

### ISSUE THREAD

**Issue by melmelissa** [linuxmint/cinnamon-spices-applets#1188]
Hello
The Disk Utilization applet does not work under Debian Sid Cinnamon version 3.2.7
With the kernel 4.12.
Minimum installation via the network in expert mode
The systat package is installed.
The applet remains blocked at 0%
When I click on it, neither my SSD nor my HDD are recognized.

Should I install a specific package that I would have forgotten ??
Or is it related to a bug ??
Or is the applet only functional with Cinnamon 3.4.4.

The return of iostat in the terminal returns me statistical data of my two hard disks.
Therefore the latter is well functional.

If a solution exists, please let me know.

thank you in advance

**Comment by melmelissa** [comment]
https://pastebin.com/d28dGMVD
https://pastebin.com/QbRNBX0L

My glass.log and xsessions-error
If it can advance the problem, or give a track to know where this bug comes from, which is really handicapped for me, since I work on heavy editing software videos and photos that use hard drives, j Likes to be able to quickly control the state live.
As for temperature, the use of cpu, ram.

If a solution or if the bug is known, please let me know.
Here, by contacting the maintainer directly, https://cinnamon-spices.linuxmint.com/applets/view/161
I did not get any answers alas ..

On Github, it is my last hope to finally understand the nature of this dysfunction, after 2 months to tear my hair by returning all forums and bug reports, on several sites ...

**Comment by oxydium** [comment]
I confirm the bug also on a Debian Buster Sid Cinnamon 3.2.7-4 with kernel 4.13
This bug was already present on Stretch.
About 4 months that it lasts ...
When is a real bug fix?
Or a clue to help us? We will be left in silence for how long?

Why make an applet that works randomly according to the distributions?
Debian always set aside in these moments ..
We're bugging up nicely hoping to be heard, and it's been 4 months since we're ignored .. not very serious on the part of a maintainer.

Proof again, on Github every bug report get answers and the only one that will pass in silence will be that one ..
Know to assume your job, or do not make applets.

And for information, the applet works well under Fedora 26 Cinnamon 3.4.6-3
@melmelissa, I understand your frustration, I also use software and I have a hard drive at the end of life, Cinnamon appeared to me rich in different applets for a system monitoring, but like what .. some applets are set aside ..

So if the applet is not compatible with Cinnamon 3.2.7, why not tell us instead of leaving us as idiots to seek, and stay in your eternal silence?
An answer is quickly said if this applet no longer works under Cinnamon 3.2.

I am exasperated, 4 months we drag this bug, and nobody shows up ..
It proves how sometimes users are put aside because they use software or other very little used by the community.
So do not do them if you are unable to assume them during a bug known and recognized for ages .. and that is manisfeste tirelessly towards you.

**Comment by melmelissa** [comment]
@oxydium
Thank you to confirm the bug, I feel less alone.
But we are not heard ... :( :(
This bug there for example https://github.com/linuxmint/cinnamon-spices-applets/issues/1190
Is already about to be resolved, an answer in less than an hour ..
As you say, the little used applets pass to the trap of the forgotten ones, and we can not even express and remount properly the bugs that exist since they make this applet unusable and therefore useless ... :( :(
By posting the bug directly on the Spice Cinnamon page, I thought I had an answer, but absolutely not, it does ghost effect ...
And it is very sad that it is we who pay the consequences ....

**Comment by jaszhix** [comment]
@oxydium
> Why make an applet that works randomly according to the distributions?
Debian always set aside in these moments ..
We're bugging up nicely hoping to be heard, and it's been 4 months since we're ignored .. not very serious on the part of a maintainer.

> Proof again, on Github every bug report get answers and the only one that will pass in silence will be that one ..
Know to assume your job, or do not make applets.

Nobody is getting paid to develop these applets. Just speaking for myself, I do this as a hobby because I enjoy it. If an applet needs attention, this is the place to report it, and from what I can tell, this issue has been open for 3 hours.

A lot of applets use dependencies that Cinnamon doesn't depend on, so there is no guarantee the library versions will match up. Its the nature of the beast, and to place blame over it is not productive in this type of forum.

The author is set to none, which means its up for grabs for anyone to contribute to. It is nobody's "job", or responsibility.

@melmelissa Please follow the [issue template](https://github.com/linuxmint/cinnamon-spices-applets/blob/master/.github/ISSUE_TEMPLATE) and provide logs. Instructions on how to obtain them are in it.

**Comment by oxydium** [comment]
No one wants to move around to help us.
We must be lucid, it is the poor truth.
Surely a bell war between the maintainer of the applet, the team of Cinnamon, against the Debian team, like ridiculous bell wars for systemd and systemV.
Eh yes cinnamon belongs to Mint.
Debian the mother company is probably the unwelcome with cinnamon.
Hence the fact that we are always in version 3.2.7 moreover ...

Oddly the applet works everywhere, except on Debian, funny coincidence.
As soon as we say that we are on Debian, nobody takes into account our bug that is serious, since this applet just does not work ..

Yes I saw the bug report, indeed this one will be corrected very quickly, yet a funny coincidence is not it.
Do not expect anything from Github and the guys from here.
They will never answer you, and will leave you in the galley with your applet that does not work ..

It has been going on for 4 months, and it is not ready to stop.

Maybe the move to Cinnamon 3.4.4 (currently in the experimental repository) will solve our worry.
And so we can conclude that the applet is not compatible with cinnamon 3.2.7
But it's too complicated for the maintainer to come and tell us directly.

At least the Debian maintainers listen to us when they are bugged.
They repair them and explain where the problem came from.

Go, good luck with them, miss

**Comment by jaszhix** [comment]
If I were maintaining this applet, I'm not sure I would want to fix it after reading these entitled ramblings, and Debian conspiracies. How about you start debugging it and submit a PR?

**Comment by mtwebster** [comment]
Let me clear something up: no one associated with Mint or Cinnamon has anything to do with package maintenance on Debian.  We do *not* officially distribute anything on Debian.

The reason it probably doesn't work in Debian, is that, as mentioned, Cinnamon is an old version.  There is no conspiracy.  We don't talk about how we can screw Debian in our spare time.   Ask your debian maintainers why Gnome, etc.. are important enough to freeze a release for weeks and months but not Cinnamon, ok?  This happened in Jessie, and it's happened again just recently.  Cinnamon works everywhere else because there are maintainers that really want Cinnamon and want it to work there and do what needs to be done to make that happen.

I have no idea if the version is even the cause here, and I honestly don't care right now.  Compatibility with an older version is a relatively low priority (though well-maintained applets tend to provide multiple code bases for different versions anyhow.

**Comment by clefebvre** [comment]
I removed quite a lot of unnecessary drama. Nobody cares about that and it gets in the way of analysis and troubleshooting. Please stick to what matters.

**Comment by clefebvre** [comment]
Hi @melmelissa,

Looking at the code... do you get a /tmp/ioDisktmp_stat.out file? And is it filled up with content?

Try to troubleshoot the applet if you can. You can use global.logError(somestring) to output content to your .xsession-errors.

**Comment by clefebvre** [comment]
@NikoKrause if nobody's maintaining this applet and it's not working with 3.2 (it's worth checking in Mint 18.1 to be sure).. assuming we fail to fix it, we should consider deleting it.

**Comment by NikoKrause** [comment]
I'm on Linux Mint 18.2 with Cinnamon 3.4.
I installed sysstat and this applet. It shows always 0% for me. Also it doesn't pop up to show my two hard drives.

So I think it's not a Debian or Cinnamon 3.2 specific issue.

To clearify a little bit. The applets on the Cinnamon spices website are 3rd party applets. This means they are created by guys, which do not belong to the Linux Mint Team.
Since all the applets from the old website have been migrated to this repo for the new website, a lot of broken and abandoned applets have been migrated as well. Some guys of the Linux Mint team have reviewed, fixed and deleted a lot of those applets. And they try to help where they can.  The author/developer of this applet has abandoned it. Some applets are easy to fix and others are difficult, especially if you are not familiar with the code. Since the guy, who is most familiar with the code, namely the author, has abandoned it, this won't be easy to fix.

If you're lucky, somebody who has time and skills sees this and might fix it or wants to maintain it. This can take some time.

The issue has been reported on the spices website 3 weeks ago. Bugs should be reported best here on GitHub. Unfortunatelly we don't get any notifications, when somebody makes a comment or reports a bug in a comment on the spices website. And nobodys clicking through hundreds of applets pages to see, if somebody wrote a new comment somewhere. So an issue is seen on GitHub faster. We know about this issue and try to improve it in the future.

That said I took a quick look at the logs and I don't see anything related to this applet. So I'm not sure what causes it. And right now I don't have much time to spare to look any further into this.

**Comment by melmelissa** [comment]
The applet is dropped.
  But it was updated 3 weeks ago, so it's recent abandonment.
But that the bug traine for 4 months .........

I posted a message on the website Spice cinnamon to try to get a reply from the maintainer himself, alas unsuccessfully.

What I do not understand is why under fedora and Arch it works well. ??
And that under Mint (some versions, since the 18.2 live usb applet works, I just tested)
And that under Debian no matter the version (Stretch, Buster) impossible to operate (symptoms described in the 1st message)

There may be a necessary dependency that works in parallel with sysstat.
Installed on Arch, Fedora, Mint 18.2, but not on a Debian in netinstall.

Damage the applet was very practical, even indispendable for me ... like all the others that I use ..

**Comment by NikoKrause** [comment]
> The applet is abandoned.??

If an applet has an active author/developer, than he's mentioned on the website, e.g. [this applet](https://cinnamon-spices.linuxmint.com/applets/view/105) says `by jamesmorgan`

And the [Disk Utilization](https://cinnamon-spices.linuxmint.com/applets/view/161) applet has no author, therefore there is no `by ...` mention.
Looking at the [old website](http://web.archive.org/web/20160603031840/https://cinnamon-spices.linuxmint.com/applets/view/161), the [author](http://web.archive.org/web/20150914160308/http://cinnamon-spices.linuxmint.com/users/view/3319) has been active last time on 2014-09-23 04:40:47

There's no notification for comments on the spices website yet. That's why nobody noticed your comment there.

> But it was updated 3 weeks ago, so it's recent abandonment.

Looking at the history/changelog of this applet by pressing on 
**Last commit:** [0fe19dc01d806dbcb5bf61594087c01ca6a1ee70](https://github.com/linuxmint/cinnamon-spices-applets/commits/master/ioDisk@ctrlesc) 
the updates on this applet have been only translations from people, who like to help out with translations.
The last update was adding a Danish translation (da.po).

**Comment by melmelissa** [comment]
In fact, looking at all the links, the applet is left for long time ..
Unfortunately I do not know anything about applet development, so I would be unable to correct it myself.
The only thing I noticed between my Archlinux and my Debian Sid is the version of sysstat.
Under Arch it is 11.4.5-1
Under Debian Sid it's 11.6.0-1
Under Fedora 26 it's 11.5.4-2

Maybe version 11.6 of sysstat triggers a bug because of the old age of the applet ... :( :(

**Comment by NikoKrause** [comment]
The applet generates and uses a file `/tmp/ioDisktmp_stat.out`, can you take screenshots of the applet on Arch, Debian, Fedora and also post the files `/tmp/ioDisktmp_stat.out` of Debian/Arch/Fedora respectively. 

As I said, I have Linux Mint 18.2 and it shows 0%. So I don't know if it works or not (therefore I'd like to see the screenshots).

**Comment by NikoKrause** [comment]
Also is this all the same hardware/machine on Debian/Arch/Fedora. (Do you have a triple boot?)

**Comment by clefebvre** [comment]
I can reproduce it here.. Mint 18.2, the .out file is populated but the applet stays at 0%.

**Comment by melmelissa** [comment]
No I have not Fedora it was to take again what Oxydium said in one of these messages.

I have a simple dual boot Arch Debian on the same machine.
I start my Arch and I capture you.
I finish my work quickly and I come back to you

**Comment by melmelissa** [comment]
Here is the screenshot under Arch, we see the applet functional.
And the log ioDisk_Tmp (a bit better filled than under Debian)

http://pix.toile-libre.org/?img=1504034959.png
https://pastebin.com/yDqD74xU

**Comment by clefebvre** [comment]
OK I can see what's wrong.

It actually works for me here... the reason I get 0% is because I'm on an SSD and it's actually almost always 0% or close to it (in the file too).

Now, the reason it doesn't work in Debian is because iostat doesn't return the same data there than in other distros. I can see in your Arch pastebin that it returns 13 values... the 13th being the percentage usage. Same thing in Mint. But looking at your pastebin in Debian I see 15 values instead... 

The code is designed to run iostat -dxh 1 2 and to gather the 13th value... so if that value isn't there but in 15th position instead, it won't work.

So the applet needs to be adapted to either run iostat in a way that works in both older and newer versions of iostat, or get the value some other way.

There. Analysis done ;)

**Comment by melmelissa** [comment]
Ha yes indeed well seen for difference.
So we know where the problem comes from.
So it is insolvent.
Debian will not change its iostat so it will be like those of Fedora, Arch etc

Very annoying of the blow :/

**Comment by NikoKrause** [comment]
Where's the Debian pastebin? I only see the Arch one.

**Comment by melmelissa** [comment]
It is here that of Debian
https://pastebin.com/tUm505Ch

**Comment by NikoKrause** [comment]
I use an SSD too^^

If it's always the last position, that's easy to fix.
@melmelissa 
Can you please try, if this changed applet version works on your machine on Debian and Arch?
Download the zip, extract it and copy the extracted folder `ioDisk@ctrlesc` to `~/.local/share/cinnamon/applets/`
[ioDisk@ctrlesc.zip](https://github.com/linuxmint/cinnamon-spices-applets/files/1261652/ioDisk.ctrlesc.zip)

**Comment by NikoKrause** [comment]
I forgot to mention, that you have to restart Cinnamon after that. Restarting Cinnamon with CTRL+ALT+ESC or just logout/restart.

**Comment by melmelissa** [comment]
Under Arch the applet still works.
I disconnected and reconnected
But under Debian it still does not work ...

**Comment by NikoKrause** [comment]
Did you restarted the machine after moving the directory?

**Comment by melmelissa** [comment]
Yes at the moment ...
And still no change ....

**Comment by NikoKrause** [comment]
The code uses a bunch of regular expressions. It relied on the fact, that iostat output has only numbers.
But the output from Debian has also stuff like `305,4k` and `5,4%`, so it uses `k` and `%` signs and not only numbers.

I don't have the time right now. Maybe I'll look in a few weeks into it or somebody else is faster than me. (Now that the reason of the bug is found.)

**Comment by xearonet** [comment]
Hi. I can take applet under my wings. 
@NikoKrause

**Comment by NikoKrause** [comment]
@xearonet feel free

**Comment by ctrlesc** [comment]
I am/was the original author.  I can probably fix it but am new to the github experience.  If someone would point me in the right direction, I would like to help and/or collaborate with xearonet.
@NikoKrause @xearonet 
Regards

**Comment by NikoKrause** [comment]
As a first start you can read this:
https://github.com/NikoKrause/CinnVIIStarkMenu/issues/28
If you run into issues, feel free to ask.

**Comment by NikoKrause** [comment]
And here https://github.com/linuxmint/cinnamon-spices-themes/blob/master/README.md#git-basics

**Comment by xearonet** [comment]
Hi @ctrlesc 
Glad to see you again.
You're still author of this applet. I want to repair It but I'm pretty busy recently.
If you find time to repair feel free to ask for help.
BR

**Comment by ctrlesc** [comment]
Thanks for welcome and help.  I got the applet working on the newest version of sysstat and submitted a pull request.
@NikoKrause @xearonet @melmelissa @oxydium

**Comment by NikoKrause** [comment]
@melmelissa The author is back and pushed a fix for it. Can you update the applet and confirm, that the issue on Debian is gone?
